src/azure-cli/azure/cli/command_modules/cognitiveservices/_params.py

```diff
@@ -147,7 +147,7 @@ def load_arguments(self, _):
         c.argument('kind', help='the API name of cognitive services account',
                    completer=kind_completer)
         c.argument('tags', tags_type)
-        c.argument('key_name', required=True, help='Key name to generate', choices=['key1', 'key2'])
+        c.argument('key_name', required=True, help='Key name to generate', choices=['Key1', 'Key2'])
```

Comment:
It will cause breaking change for existing command users. Please wrapper it with CaseInsensitiveList